### PR TITLE
[GSoC 2015 MRG] Restructuring Montecarlo

### PR DESCRIPTION
**Pull request by kaushik94**
This is still unfinished and in an embarrassing state. TODO - split the header files accordingly and all other usages of the functions in python files have to he changes I guess. #293.

### CODE REVIEW

**Comment by orbitfold** [comment]
I think the best way to structure the code is to let the structure and the methods that operate on that structure to be in the same file. Think how you would structure a large C++ program. It really does not make sense for some functions that operate on an rpacket to sit in one file and others in another. I propose to make a file called rpacket.c to hold the methods and rpacket.h to hold the data structure. storage.h should hold the storage structure and utilities.c and utilities.h should hold the search functions and other stuff. It is really important that getter and setter methods go in the same file as other functions that operate on an rpacket. Basically think of it as a C++ class and put all the "methods" in the same file. Now it is arbitrary and confusing IMO.

**Comment by kaushik94** [comment]
@orbitfold  It was very dumb on my part. I have to clearly know first what these functions do, so will do it in a while. About a week or so for the finished one. Check out the new one, better ?

File: tardis/montecarlo/src/search_routines.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#include "search_routines.h"
```
Comment (wkerzendorf) [inline tardis/montecarlo/src/search_routines.c]: maybe put this into helpers or call it `utils.c`. @orbitfold what do you think?

**Comment by wkerzendorf** [comment]
@kaushik94 no worries. It's all a learning process :wink:

**Comment by kaushik94** [comment]
Hey @wkerzendorf  and @orbitfold  can you please take a look at it and let me know if thats in the lines of what you are thinking. Now we can create `geometry.c` and `utils.h` and I think we'll be done with the restructuring ?

**Comment by wkerzendorf** [comment]
@kaushik94 it looks like a good start. why did you change the access from 644->755?

**Comment by kaushik94** [comment]
@wkerzendorf  even I am wondering. I took a new mac, its my first time working on OSX. I thing I messed up with the configuration or something. Anyway to revert the permissions back ?

**Comment by wkerzendorf** [comment]
hmm - use chmod. that should work.

**Comment by kaushik94** [comment]
Yes, I think that worked !

**Comment by wkerzendorf** [comment]
looks good!

**Comment by kaushik94** [comment]
I think we have to change the name of `cmontecarlo` to `utils` or `geometry` or both. But just waiting for @orbitfold to review, know what he thinks. Anything more we should do ?

**Comment by wkerzendorf** [comment]
@kaushik94 - not really except maybe makeing a new PR from that that tries to test the rpacket stuff. That should be fairly simple.

**Comment by kaushik94** [comment]
Yes I'll get to that by Tuesday probably. I'll get done with this one, write documentation, remove unused imports etc. Shall we go with the testing once this is merged or you want me make it in parallel?

**Comment by wkerzendorf** [comment]
@orbitfold can you have a quick look and tell us if it is ready to be merged. @ssim if you get a chance please have a look. I will merge by monday if there are no serious issues.

**Comment by wkerzendorf** [comment]
@mklauser please have a look as well.

**Pull request merged**